File: androidApp/src/main/java/com/app/pustakam/android/screen/settings/SettingsView.kt
package com.app.pustakam.android.screen.settings

import com.app.pustakam.android.theme.GranthCopper
import com.app.pustakam.android.theme.GranthForest
import com.app.pustakam.android.theme.GranthGold
import com.app.pustakam.android.theme.GranthIndigo
import com.app.pustakam.android.theme.GranthIvory
import com.app.pustakam.android.theme.GranthSaffron
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.KeyboardArrowRight
import androidx.compose.material.icons.automirrored.filled.MenuBook
import com.app.pustakam.android.screen.notebookReader.ReadingMode
import androidx.compose.material.icons.filled.Accessibility
import androidx.compose.material.icons.filled.CloudQueue
import androidx.compose.material.icons.filled.History
import androidx.compose.material.icons.filled.Keyboard
import androidx.compose.material.icons.filled.Language
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material.icons.filled.NightlightRound
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material.icons.filled.SwapHoriz
import androidx.compose.material.icons.filled.SystemUpdateAlt
import androidx.compose.material.icons.filled.TextFields
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.MaterialTheme.colorScheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Switch
import androidx.compose.material3.SwitchDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import androidx.lifecycle.viewmodel.compose.viewModel
import com.app.pustakam.android.MyApplicationTheme
import com.app.pustakam.android.screen.navigation.Route
import com.app.pustakam.android.widgets.CircleIconLoad
import com.app.pustakam.core.network.toAbsoluteMediaUrl
import com.app.pustakam.android.theme.ThemeMode
import com.app.pustakam.android.theme.ThemeTilePicker
import com.app.pustakam.android.theme.radiusLg
import com.app.pustakam.android.theme.radiusMd

// 🎨 22-Jul-2026 — Granth spec §6 Settings, matched to iOS SettingsView.swift. Replaces the
//   placeholder Text("Settings") with the profile header + grouped sections. Appearance is live-wired
//   through SettingsViewModel → shared DataStore; the remaining rows are presentational until their
//   features land.
@Composable
fun SettingsScreen(
    // 🔧 22-Jul-2026 — was `(Route)->Unit` importing okhttp3.Route by mistake; navigateTo takes a String
    onNavigate: (String) -> Unit,
    onBack: () -> Unit = {},
    viewModel: SettingsViewModel = viewModel()
) {
    val state by viewModel.uiState.collectAsStateWithLifecycle()

    SettingsContent(
        state = state,
        onThemeModeSelected = viewModel::onThemeModeSelected,
        onMarkdownShortcutsChange = viewModel::onMarkdownShortcutsChange,
        onFocusModeDimmingChange = viewModel::onFocusModeDimmingChange,
        onCloudSyncChange = viewModel::onCloudSyncChange,
        onAutoBackupChange = viewModel::onAutoBackupChange,
        onOfflineModeChange = viewModel::onOfflineModeChange,
        onReadingModeChange = viewModel::onReadingModeChange,
        onOpenProfile = { onNavigate(Route.Profile) },
        onLogout = viewModel::logout
    )
}

// 🎨 22-Jul-2026 — stateless body so previews can drive it with sample data (no ViewModel/Koin needed).
@Composable
fun SettingsContent(
    state: SettingsUiState,
    onThemeModeSelected: (ThemeMode) -> Unit = {},
    onMarkdownShortcutsChange: (Boolean) -> Unit = {},
    onFocusModeDimmingChange: (Boolean) -> Unit = {},
    onCloudSyncChange: (Boolean) -> Unit = {},
    onAutoBackupChange: (Boolean) -> Unit = {},
    onOfflineModeChange: (Boolean) -> Unit = {},
    onReadingModeChange: (Boolean) -> Unit = {},
    onOpenProfile: () -> Unit = {},
    onLogout : () -> Unit = {}
) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(colorScheme.background)
            .verticalScroll(rememberScrollState())
            .padding(horizontal = 20.dp, vertical = 16.dp),
        verticalArrangement = Arrangement.spacedBy(24.dp)
    ) {

        SettingsProfileHeader(
            initial = state.profileInitial,
            name = state.profileName,
            subtitle = state.profileSubtitle,
            badge = state.profileBadge,
            avatarUrl = state.user?.avatarUrl.toAbsoluteMediaUrl(),
            onClick = onOpenProfile
        )

        // 🎨 Appearance — the only live section
        SettingsSection(title = "Appearance") {
            Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
                ThemeTilePicker(
                    selection = state.themeMode,
                    onSelect = onThemeModeSelected
                )
                Text(
                    text = appearanceHint(state.themeMode),
                    style = MaterialTheme.typography.bodySmall,
                    color = colorScheme.outline
                )
            }
        }

        SettingsSection(title = "Editor & Reading") {
            SettingsCard {
                SettingsRow(
                    icon = Icons.Default.TextFields, tint = GranthSaffron,
                    title = "Editor font", subtitle = state.editorFontSubtitle,
                    value = state.editorFont
                )
                SettingsRow(
                    icon = Icons.Default.Keyboard, tint = GranthForest,
                    title = "Markdown shortcuts", subtitle = "Auto-format as you type",
                    isOn = state.markdownShortcuts, onCheckedChange = onMarkdownShortcutsChange
                )
                SettingsRow(
                    icon = Icons.Default.NightlightRound, tint = GranthCopper,
                    title = "Focus mode dimming", subtitle = "Fade inactive paragraphs",
                    isOn = state.focusModeDimming, onCheckedChange = onFocusModeDimmingChange
                )
                // 📖 23-Jul-2026: NEW — reader layout. Same persisted value as the reader's toolbar
                //   toggle, so changing it in either place updates the other.
                SettingsRow(
                    icon = Icons.AutoMirrored.Filled.MenuBook, tint = GranthGold,
                    title = "Scrolling mode",
                    subtitle = if (state.readingMode == ReadingMode.SCROLL)
                        "Documents scroll continuously" else "Documents turn like book pages",
                    isOn = state.readingMode == ReadingMode.SCROLL,
                    onCheckedChange = onReadingModeChange
                )
                SettingsRow(
                    icon = Icons.Default.History, tint = GranthIndigo,
                    title = "Version history", subtitle = state.versionHistorySubtitle,
                    value = "", isLast = true
                )
            }
        }

        SettingsSection(title = "Sync & Backup") {
            SettingsCard {
                SettingsRow(
                    icon = Icons.Default.CloudQueue, tint = GranthForest,
                    title = "Cloud sync", subtitle = state.cloudSyncSubtitle,
                    isOn = state.cloudSync, onCheckedChange = onCloudSyncChange
                )
                SettingsRow(
                    icon = Icons.Default.SystemUpdateAlt, tint = GranthSaffron,
                    title = "Auto backup", subtitle = "Daily · encrypted",
                    isOn = state.autoBackup, onCheckedChange = onAutoBackupChange
                )
                SettingsRow(
                    icon = Icons.Default.SwapHoriz, tint = GranthCopper,
                    title = "Offline mode", subtitle = "Keep all notes on device",
                    isOn = state.offlineMode, onCheckedChange = onOfflineModeChange, isLast = true
                )
            }
        }

        SettingsSection(title = "More") {
            SettingsCard {
                SettingsRow(
                    icon = Icons.Default.Language, tint = GranthIndigo,
                    title = "Language", value = state.language
                )
                SettingsRow(
                    icon = Icons.Default.Accessibility, tint = GranthGold,
                    title = "Accessibility", value = ""
                )
                SettingsRow(
                    icon = Icons.Default.Lock, tint = GranthCopper,
                    title = "Privacy & Security", subtitle = "App lock · Face ID", value = ""
                )
                SettingsRow(
                    icon = Icons.Default.Settings, tint = GranthForest,
                    title = "Developer options", value = "",
                )
                SettingsRow(
                    icon = Icons.Default.Settings, tint = Color.Red,
                    title = "Logout", value = "",
                    onClick = onLogout
                )
            }
        }
    }
}

// 🎨 tells the user what the current pick actually does (matches iOS copy)
private fun appearanceHint(mode: ThemeMode): String = when (mode) {
    ThemeMode.SYSTEM -> "Follows your device appearance, including scheduled Dark Mode."
    ThemeMode.LIGHT -> "Always uses the warm parchment palette."
    ThemeMode.DARK -> "Always uses the warm dark-ink palette."
    ThemeMode.AMOLED -> "True black surfaces — saves power on OLED displays."
}

// MARK: - Building blocks

// 🎨 22-Jul-2026 — uppercase group label + content, matching spec §6 grouping.
@Composable
fun SettingsSection(title: String, content: @Composable () -> Unit) {
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        Text(
            text = title.uppercase(),
            style = MaterialTheme.typography.labelSmall,
            fontWeight = FontWeight.Bold,
            letterSpacing = 1.2.sp,
            color = colorScheme.outline,
            modifier = Modifier.padding(horizontal = 4.dp)
        )
        content()
    }
}

// 🎨 22-Jul-2026 — rounded surface that hosts a run of SettingsRow.
@Composable
fun SettingsCard(content: @Composable () -> Unit) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(radiusMd))
            .background(colorScheme.surface)
            .border(1.dp, colorScheme.outlineVariant, RoundedCornerShape(radiusMd))
    ) { content() }
}

// 🎨 22-Jul-2026 — one settings row: tinted icon, title/subtitle, then a switch OR a value chevron.
@Composable
fun SettingsRow(
    icon: ImageVector,
    tint: Color,
    title: String,
    subtitle: String? = null,
    value: String? = null,
    isOn: Boolean? = null,
    onCheckedChange: (Boolean) -> Unit = {},
    onClick: () -> Unit = {},
    isLast: Boolean = false
) {
    Column {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .then(if (isOn == null) Modifier.clickable(onClick = onClick) else Modifier)
                .padding(horizontal = 16.dp, vertical = 12.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            Box(
                modifier = Modifier
                    .size(32.dp)
                    .clip(RoundedCornerShape(9.dp))
                    .background(tint.copy(alpha = 0.16f)),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    imageVector = icon,
                    contentDescription = null,
                    tint = tint,
                    modifier = Modifier.size(18.dp)
                )
            }

            Column(modifier = Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(2.dp)) {
                Text(
                    text = title,
                    style = MaterialTheme.typography.bodyMedium,
                    color = colorScheme.onSurface
                )
                if (subtitle != null) {
                    Text(
                        text = subtitle,
                        style = MaterialTheme.typography.bodySmall,
                        color = colorScheme.outline,
                        maxLines = 1,
                        overflow = TextOverflow.Ellipsis
                    )
                }
            }

            if (isOn != null) {
                Switch(
                    checked = isOn,
                    onCheckedChange = onCheckedChange,
                    colors = SwitchDefaults.colors(
                        checkedTrackColor = colorScheme.primary,
                        checkedThumbColor = GranthIvory
                    )
                )
            } else {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    if (!value.isNullOrEmpty()) {
                        Text(
                            text = value,
                            style = MaterialTheme.typography.bodySmall,
                            color = colorScheme.outline
                        )
                        Spacer(Modifier.width(4.dp))
                    }
                    Icon(
                        imageVector = Icons.AutoMirrored.Filled.KeyboardArrowRight,
                        contentDescription = null,
                        tint = colorScheme.outline,
                        modifier = Modifier.size(18.dp)
                    )
                }
            }
        }

        // 🎨 inset divider aligns under the text, not the icon (matches iOS 60pt leading inset)
        if (!isLast) {
            HorizontalDivider(
                modifier = Modifier.padding(start = 60.dp),
                color = colorScheme.outlineVariant
            )
        }
    }
}

// 🎨 22-Jul-2026 — profile header (spec §6): gradient avatar, identity, PRO badge.
@Composable
fun SettingsProfileHeader(
    initial: String,
    name: String,
    subtitle: String,
    badge: String? = null,
    avatarUrl: String? = null,
    onClick: () -> Unit = {}
) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(radiusLg))
            .background(colorScheme.surface)
            .border(1.dp, colorScheme.outlineVariant, RoundedCornerShape(radiusLg))
            .clickable(onClick = onClick)
            .padding(16.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(14.dp)
    ) {
        // 🖼️ the gradient initial is the placeholder, so a user with no picture still reads as themselves
        if (avatarUrl.isNullOrBlank()) {
            Box(
                modifier = Modifier
                    .size(54.dp)
                    .clip(CircleShape)
                    .background(Brush.linearGradient(listOf(GranthIndigo, GranthForest))),
                contentAlignment = Alignment.Center
            ) {
                Text(
                    text = initial,
                    style = MaterialTheme.typography.displaySmall,
                    color = Color.White
                )
            }
        } else {
            CircleIconLoad(url = avatarUrl, size = 54.dp, onClick = onClick)
        }

        Column(modifier = Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(3.dp)) {
            Text(
                text = name,
                style = MaterialTheme.typography.headlineSmall,
                color = colorScheme.onSurface
            )
            Text(
                text = subtitle,
                style = MaterialTheme.typography.bodySmall,
                color = colorScheme.outline,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis
            )
        }

        if (badge != null) {
            Text(
                text = badge,
                style = MaterialTheme.typography.labelSmall,
                fontWeight = FontWeight.Black,
                letterSpacing = 0.8.sp,
                color = Color(0xFF7E3F20),
                modifier = Modifier
                    .clip(RoundedCornerShape(8.dp))
                    .background(Brush.linearGradient(listOf(GranthSaffron, GranthGold)))
                    .padding(horizontal = 9.dp, vertical = 5.dp)
            )
        }
    }
}

// MARK: - Previews (sample data)

// 🎨 22-Jul-2026 — sample-data previews, one per theme mode (project convention).
private val sampleSettingsState = SettingsUiState(themeMode = ThemeMode.LIGHT)

@Preview(name = "Settings — Light", showBackground = true, heightDp = 1200)
@Composable
private fun SettingsLightPreview() {
    MyApplicationTheme(isDarkTheme = false) {
        Surface(color = colorScheme.background) {
            SettingsContent(state = sampleSettingsState)
        }
    }
}

@Preview(name = "Settings — Dark", showBackground = true, heightDp = 1200)
@Composable
private fun SettingsDarkPreview() {
    MyApplicationTheme(isDarkTheme = true) {
        Surface(color = colorScheme.background) {
            SettingsContent(state = sampleSettingsState.copy(themeMode = ThemeMode.DARK))
        }
    }
}

@Preview(name = "Settings — AMOLED", showBackground = true, heightDp = 1200)
@Composable
private fun SettingsAmoledPreview() {
    MyApplicationTheme(isDarkTheme = true, isAmoled = true) {
        Surface(color = colorScheme.background) {
            SettingsContent(state = sampleSettingsState.copy(themeMode = ThemeMode.AMOLED))
        }
    }
}

File: iosApp/iosApp/view/screens/settings/SettingsView.swift
import SwiftUI

struct SettingsView: View {

    @Environment(ThemeManager.self) private var themeManager
    @Environment(\.palette) private var palette
    @Environment(Router.self) var router: Router
    @State private var readerPrefs = ReaderPrefsAdapter()
    @State private var readingMode: ReadingMode = .page
    @State private var authBridge = AuthBridgeAdapter()
    // 👤 31-Aug-2026 — the header used to be four hardcoded strings, one of them an email address
    @StateObject private var profile = ProfileViewModel()
    var body: some View {
        @Bindable var themeManager = themeManager

        ScrollView {
            VStack(alignment: .leading, spacing: Theme.Spacing.xxl) {

                SettingsProfileHeader(
                    initial: profile.initial,
                    name: profile.user == nil ? "Your profile" : profile.displayName,
                    subtitle: profileSubtitle,
                    badge: profileBadge,
                    avatarUrl: profile.avatarUrl,
                    onTap: { router.navigate(to: .Profile) }
                )

                // 🎨 Appearance — the only live section
                SettingsSection(title: "Appearance") {
                    VStack(alignment: .leading, spacing: Theme.Spacing.md) {
                        ThemeTilePicker(selection: $themeManager.mode)

                        Text(appearanceHint)
                            .font(Theme.Fonts.metaCaption)
                            .foregroundStyle(palette.text3)
                            .fixedSize(horizontal: false, vertical: true)
                    }
                }

                SettingsSection(title: "Editor & Reading") {
                    SettingsCard {
                        SettingsRow(icon: "textformat", tint: Theme.Colors.saffron,
                                    title: "Editor font", subtitle: "Display serif · 16pt", value: "Iowan")
                        SettingsRow(icon: "keyboard", tint: Theme.Colors.forest,
                                    title: "Markdown shortcuts", subtitle: "Auto-format as you type", isOn: true)
                        SettingsRow(icon: "moonphase.waning.crescent", tint: Theme.Colors.copper,
                                    title: "Focus mode dimming", subtitle: "Fade inactive paragraphs", isOn: true)
                        // 📖 23-Jul-2026 — reader layout. Writes the SAME persisted value as the
                        //   reader's toolbar toggle, so changing it in either place updates the other.
                        SettingsRow(icon: "doc.plaintext", tint: Theme.Colors.gold,
                                    title: "Scrolling mode",
                                    subtitle: readingMode == .scroll
                                        ? "Documents scroll continuously"
                                        : "Documents turn like book pages",
                                    isOn: readingMode == .scroll,
                                    onToggle: { isOn in
                                        let next: ReadingMode = isOn ? .scroll : .page
                                        readingMode = next
                                        readerPrefs.setReadingMode(next)
                                    })
                        SettingsRow(icon: "clock.arrow.circlepath", tint: Theme.Colors.indigo,
                                    title: "Version history", subtitle: "Keep 30 days of snapshots",
                                    value: "", isLast: true)
                    }
                }

                SettingsSection(title: "Sync & Backup") {
                    SettingsCard {
                        SettingsRow(icon: "icloud.fill", tint: Theme.Colors.forest,
                                    title: "Cloud sync", subtitle: "Last synced 2 min ago · 1.2 GB", isOn: true)
                        SettingsRow(icon: "arrow.down.circle.fill", tint: Theme.Colors.saffron,
                                    title: "Auto backup", subtitle: "Daily · encrypted", isOn: true)
                        SettingsRow(icon: "arrow.left.arrow.right", tint: Theme.Colors.copper,
                                    title: "Offline mode", subtitle: "Keep all notes on device",
                                    isOn: false, isLast: true)
                    }
                }

                SettingsSection(title: "More") {
                    SettingsCard {
                        SettingsRow(icon: "globe", tint: Theme.Colors.indigo,
                                    title: "Language", value: "English")
                        SettingsRow(icon: "figure.wave", tint: Theme.Colors.gold,
                                    title: "Accessibility", value: "")
                        SettingsRow(icon: "lock.fill", tint: Theme.Colors.copper,
                                    title: "Privacy & Security", subtitle: "App lock · Face ID", value: "")
                        SettingsRow(icon: "gearshape.2.fill", tint: Theme.Colors.forest,
                                    title: "Developer options", value: "",)
                        SettingsRow(icon: "rectangle.portrait.and.arrow.right", tint: Color.red,
                                    title: "Logout ", value: "", onTap: { authBridge.logout {router.navigate(to: .Login)}})
                    }
                }
            }
            .padding(.horizontal, Theme.Spacing.gutter)
            .padding(.vertical, Theme.Spacing.lg)
        }
        .background(palette.background.ignoresSafeArea())
        // 📖 23-Jul-2026 — restore the persisted pick, and follow changes made in the reader
        .onAppear { readerPrefs.observeReadingMode { readingMode = $0 } }
        .task { profile.load() }
    }

    /// 🔒 The handle, never the email. Email and phone are login credentials from 31-Aug-2026 and
    /// the server no longer returns anyone else's; rendering our own here would still teach the
    /// wrong habit, and the handle is what a person actually shares.
    private var profileSubtitle: String {
        guard profile.user != nil else { return "Tap to set up" }
        return profile.handle.isEmpty ? "Tap to pick a username" : profile.handle
    }

    /// 🆔 the nudge replaces the badge until a real handle exists.
    private var profileBadge: String? {
        guard let user = profile.user else { return nil }
        return (user.username ?? "").isEmpty ? "SET UP" : nil
    }

    // 🎨 tells the user what the current pick actually does
    private var appearanceHint: String {
        switch themeManager.mode {
        case .system: return "Follows your iOS appearance, including scheduled Dark Mode."
        case .light:  return "Always uses the warm parchment palette."
        case .dark:   return "Always uses the warm dark-ink palette."
        case .amoled: return "True black surfaces — saves power on OLED displays."
        }
    }
}

// MARK: - Building blocks

// 🎨 22-Jul-2026 — uppercase group label + content, matching spec §6 grouping.
struct SettingsSection<Content: View>: View {
    let title: String
    @ViewBuilder var content: Content
    @Environment(\.palette) private var palette

    var body: some View {
        VStack(alignment: .leading, spacing: Theme.Spacing.sm) {
            Text(title.uppercased())
                .font(Theme.Fonts.metaCaption)
                .tracking(1.2)
                .foregroundStyle(palette.text3)
                .padding(.horizontal, Theme.Spacing.xs)
            content
        }
    }
}

// 🎨 22-Jul-2026 — rounded surface that hosts a run of SettingsRow.
struct SettingsCard<Content: View>: View {
    @ViewBuilder var content: Content
    @Environment(\.palette) private var palette

    var body: some View {
        VStack(spacing: 0) { content }
            .background(palette.surface)
            .clipShape(RoundedRectangle(cornerRadius: Theme.Radius.md))
            .overlay(
                RoundedRectangle(cornerRadius: Theme.Radius.md)
                    .stroke(palette.border, lineWidth: 1)
            )
            .shadow(color: Theme.Elevation.card,
                    radius: Theme.Elevation.cardRadius, x: 0, y: Theme.Elevation.cardY)
    }
}

// 🎨 22-Jul-2026 — one settings row: tinted icon, title/subtitle, then a toggle OR a value chevron.
struct SettingsRow: View {
    let icon: String
    let tint: Color
    let title: String
    var subtitle: String? = nil
    var value: String? = nil
    var isOn: Bool? = nil
    // 📖 23-Jul-2026 — rows that own real state pass a callback; presentational rows omit it
    var onToggle: ((Bool) -> Void)? = nil
    var isLast: Bool = false
    var onTap: () -> Void  = {}
    @State private var toggleState: Bool = false
    @Environment(\.palette) private var palette

    var body: some View {
        VStack(spacing: 0) {
            HStack(spacing: Theme.Spacing.md) {

                RoundedRectangle(cornerRadius: 9)
                    .fill(tint.opacity(0.16))
                    .frame(width: 32, height: 32)
                    .overlay(
                        Image(systemName: icon)
                            .font(.system(size: 14, weight: .semibold))
                            .foregroundStyle(tint)
                    )

                VStack(alignment: .leading, spacing: 2) {
                    Text(title)
                        .font(Theme.Fonts.bodyText)
                        .foregroundStyle(palette.text)
                    if let subtitle {
                        Text(subtitle)
                            .font(Theme.Fonts.metaCaption)
                            .foregroundStyle(palette.text3)
                    }
                }

                Spacer(minLength: Theme.Spacing.sm)
                if isOn != nil {
                    Toggle("", isOn: $toggleState)
                        .labelsHidden()
                        .tint(palette.accent)
                        // 📖 23-Jul-2026 — forward changes to the owner when the row is state-backed
                        .onChange(of: toggleState) { _, newValue in onToggle?(newValue) }
                } else {
                    HStack(spacing: 4) {
                        if let value, !value.isEmpty {
                            Text(value)
                                .font(Theme.Fonts.metaCaption)
                                .foregroundStyle(palette.text3)
                        }
                        Image(systemName: Images.arrowRight)
                            .font(.system(size: 12, weight: .semibold))
                            .foregroundStyle(palette.text3)
                    }
                }
            }
            .padding(.horizontal, Theme.Spacing.lg)
            .padding(.vertical, Theme.Spacing.md)

            if !isLast {
                Divider()
                    .overlay(palette.border)
                    .padding(.leading, 60)
            }
        }
        .contentShape(Rectangle())
        .onAppear { toggleState = isOn ?? false }
        .onTapGesture {
            onTap()
        }
        // 📖 23-Jul-2026 — follow the owner's value when it changes elsewhere (e.g. the reader's
        //   toolbar toggle). Assigning the same value doesn't re-fire onChange, so no write echo.
        .onChange(of: isOn) { _, newValue in
            if let newValue, newValue != toggleState { toggleState = newValue }
        }
    }
}

// 🎨 22-Jul-2026 — profile header (spec §6): gradient avatar, identity, PRO badge.
struct SettingsProfileHeader: View {

    let initial: String
    let name: String
    let subtitle: String
    let badge: String?
    var avatarUrl: String? = nil
    var onTap: () -> Void = {}

    @Environment(\.palette) private var palette

    var body: some View {
        HStack(spacing: Theme.Spacing.lg) {

            // 🖼️ the gradient initial is the placeholder, so a user with no picture still reads as themselves
            if let avatarUrl, !avatarUrl.isEmpty, let url = URL(string: avatarUrl) {
                AsyncImage(url: url) { phase in
                    if let image = phase.image {
                        image.resizable().scaledToFill()
                    } else {
                        gradientInitial
                    }
                }
                .frame(width: 54, height: 54)
                .clipShape(Circle())
            } else {
                gradientInitial
            }

            VStack(alignment: .leading, spacing: 3) {
                Text(name)
                    .font(Theme.Fonts.sectionTitle)
                    .foregroundStyle(palette.text)
                Text(subtitle)
                    .font(Theme.Fonts.metaCaption)
                    .foregroundStyle(palette.text3)
                    .lineLimit(1)
            }

            Spacer(minLength: Theme.Spacing.sm)

            if let badge {
                Text(badge)
                    .font(.system(size: 10, weight: .heavy))
                    .tracking(0.8)
                    .foregroundStyle(Color(hex: "#7E3F20"))
                    .padding(.horizontal, 9)
                    .padding(.vertical, 5)
                    .background(
                        LinearGradient(colors: [Theme.Colors.saffron, Theme.Colors.gold],
                                       startPoint: .topLeading, endPoint: .bottomTrailing)
                    )
                    .clipShape(RoundedRectangle(cornerRadius: Theme.Radius.sm))
            }
        }
        .padding(Theme.Spacing.lg)
        .background(palette.surface)
        .clipShape(RoundedRectangle(cornerRadius: Theme.Radius.lg))
        .overlay(
            RoundedRectangle(cornerRadius: Theme.Radius.lg)
                .stroke(palette.border, lineWidth: 1)
        )
        .shadow(color: Theme.Elevation.card,
                radius: Theme.Elevation.cardRadius, x: 0, y: Theme.Elevation.cardY)
        .contentShape(Rectangle())
        .onTapGesture { onTap() }
    }

    private var gradientInitial: some View {
        Circle()
            .fill(LinearGradient(colors: [Theme.Colors.indigo, Theme.Colors.forest],
                                 startPoint: .topLeading, endPoint: .bottomTrailing))
            .frame(width: 54, height: 54)
            .overlay(
                Text(initial)
                    .font(Theme.Fonts.displayL)
                    .foregroundStyle(.white)
            )
    }
}

// MARK: - Previews (sample data)

#Preview("Settings — Light") {
    SettingsPreviewHost(mode: .light).preferredColorScheme(.light)
}

#Preview("Settings — Dark") {
    SettingsPreviewHost(mode: .dark).preferredColorScheme(.dark)
}

#Preview("Settings — AMOLED") {
    SettingsPreviewHost(mode: .amoled).preferredColorScheme(.dark)
}

// 🎨 22-Jul-2026 — preview host injects ThemeManager + palette like the app root does.
private struct SettingsPreviewHost: View {
    let mode: ThemeMode
    @State private var manager = ThemeManager()
    @Environment(\.colorScheme) private var scheme

    var body: some View {
        SettingsView()
            .environment(manager)
            .environment(\.palette, ThemePalette(mode: mode, scheme: scheme))
            .onAppear { manager.setMode(mode) }
    }
}
